### PR TITLE
DevX newsletter: June edition

### PR DESCRIPTION
**Pull request by bobheadxi**
https://deploy-preview-3884--sourcegraph-handbook.netlify.app/departments/engineering/dev/enablement/dev-experience/newsletter/

### CODE REVIEW

File: content/departments/engineering/dev/enablement/dev-experience/newsletter.md [hunk 1]
```diff
@@ -8,6 +8,50 @@ To learn more about components of Sourcegraph's developer experience, check out
...
+
+### CLA bot automation
+
+The process of ensuring an external contributor has signed Sourcegraph's Contributor License Agreement (CLA) is now fully automated. Once a contributor has signed [the CLA form](https://docs.google.com/forms/d/e/1FAIpQLSfxy_9WJptKeTmTsrQ6C-5JeiVs4i1pUiahzgLZta1t6Nls-g/viewform), their provided GitHub handle will now automatically be synchronized to the list of approved users.
+
+You can learn more in the [`clabot-config` repository](https://github.com/sourcegraph/clabot-config) and the [accepting external contributions guide](https://docs.sourcegraph.com/dev/contributing/accepting_contribution).
+
+### `sg` goodies 
+
+**`sg [cmd...] --feedback`**: TODO @burmudar
```
Comment (jhchabran) [inline content/departments/engineering/dev/enablement/dev-experience/newsletter.md]: cc @burmudar !

File: content/departments/engineering/dev/enablement/dev-experience/newsletter.md [hunk 2]
```diff
@@ -8,6 +8,62 @@ To learn more about components of Sourcegraph's developer experience, check out
...
+
+**Generated `sg` documentation**: Because maintaining documentation is always hard, what's better than automation to make sure its stays up to date? The [`sg` reference](https://docs.sourcegraph.com/dev/background-information/sg/reference) is now automatically generated. 
+
+**`dev/schemadoc`** has been removed and is replaced by `sg migration ...` commands. Example: `sg migration describe -db codeintel --format=psql -force -out internal/database/schema.codeintel.md` to generate the schema for the `codeintel` db thanks to [#35905](https://github.com/sourcegraph/sourcegraph/pull/35905).
```
Comment (jhchabran) [inline content/departments/engineering/dev/enablement/dev-experience/newsletter.md:71]: cc @efritz if you want to edit/add something there :)

File: content/departments/engineering/dev/enablement/dev-experience/newsletter.md [hunk 3]
```diff
@@ -8,6 +8,66 @@ To learn more about components of Sourcegraph's developer experience, check out
...
+
+**Readability improvements**: `sg start logs` are now easier to read, as the command names text is now justified. To ensure it's still readable in small terminals, a few of them have be shortened and the `enterprise-` prefix is now implicit whereas `oss-` prefix has been introduced.
+
+![](https://user-images.githubusercontent.com/23356519/174646815-843dbbf0-c4e2-49a1-b046-cc3e75f047f7.png)
+
+### CI improvements
+
+The linter job that runs on every build is now inferring which linter task needs to run depending on the changes (except `main` where it runs everything), saving some time on pull requests thanks to [#35331](https://github.com/sourcegraph/sourcegraph/pull/35331).
+
+### Tech Radar
```
Comment (jhchabran) [inline content/departments/engineering/dev/enablement/dev-experience/newsletter.md:93]: I think we'll give more visibility and engagement with the radar if we mention it every month. Ideally, it'd be great if team would update it so we'd see the _new_ tags, but that's good enough as it is. cc @vrto

Comment (vrto) [reply content/departments/engineering/dev/enablement/dev-experience/newsletter.md:93]: Good idea, thanks @jhchabran

File: content/departments/engineering/dev/enablement/dev-experience/newsletter.md [hunk 4]
```diff
@@ -8,6 +8,83 @@ To learn more about components of Sourcegraph's developer experience, check out
...
+
+**Generated `sg` documentation**: Because maintaining documentation is always hard, what's better than automation to make sure its stays up to date? The [`sg` reference](https://docs.sourcegraph.com/dev/background-information/sg/reference) is now automatically generated.
+
+**`sg setup` and `sg lint` rewrite**: We have rewritten `sg lint` and `sg setup` to use a shared internal framework that can be used to easily and flexibly build powerful "check-and-fix" tasks. This enables features like:
+
+- `sg setup -check` to quickly generate a report of what you have set up
+- `sg setup -fix` to fix all issues with your dev setup in one go
+- `sg lint -fix` to automatically try and fix your lint issues (only supported by a few linters at the moment, but more can easily be added!)
```
Comment (bobheadxi) [inline content/departments/engineering/dev/enablement/dev-experience/newsletter.md:68]: Relies on https://github.com/sourcegraph/sourcegraph/pull/37528

**Comment by jhchabran** [comment]
@bobheadxi don't forget to add the loom video from https://sourcegraph.slack.com/archives/C01N83PS4TU/p1655998549750479 !

**Pull request merged**